src/printer.rs

```diff
@@ -24,12 +31,27 @@ impl<'a> Printer<'a> {
             Colors::plain()
         };
 
-        Printer {
+        // Create the instance.
+        let mut instance = Printer {
             handle,
             colors,
             config,
+            panel_width: 0,
             line_changes: None,
-        }
+        };
+
+        // Generate the panel (gutter) width.
```

Comment:
Could this be computed first and then simply passed to the `Printer` constructor? This way, we don't need the (mutable) `instance` variable.

Reply:
I was thinking that it would be a fair tradeoff, making the code easier to maintain by only having a single function (`gen_decorations`) for determining which components to display and their sizes.

I could change it with a couple of `if`s and consts, if you prefer to avoid the mutable `instance` variable though.